homeassistant/components/lutron_caseta/config_flow.py

```diff
@@ -31,22 +49,129 @@ class LutronCasetaFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+                return self.async_create_entry(title=self.bridge_id, data=self.data)
+
+            assets = None
+            try:
+                assets = await async_pair(self.data[CONF_HOST])
+            except (asyncio.TimeoutError, OSError):
+                errors["base"] = "cannot_connect"
+
+            if not errors:
+                await self.hass.async_add_executor_job(self._write_tls_assets, assets)
+                return self.async_create_entry(title=self.bridge_id, data=self.data)
+
+        return self.async_show_form(
+            step_id="link",
+            errors=errors,
+            description_placeholders={
+                CONF_NAME: self.bridge_id,
+                CONF_HOST: self.data[CONF_HOST],
+            },
+        )
+
+    @property
+    def bridge_id(self):
+        """Return the best identifier for the bridge.
+
+        If the bridge was not discovered via zeroconf,
+        we fallback to using the host.
+        """
+        return self.lutron_id or self.data[CONF_HOST]
+
+    def _write_tls_assets(self, assets):
+        """Write the tls assets to disk."""
+        for asset_key, conf_key in FILE_MAPPING.items():
+            with open(self.hass.config.path(self.data[conf_key]), "w") as file_handle:
+                file_handle.write(assets[asset_key])
+
+    def _tls_assets_exist(self):
+        """Check to see if tls assets are already on disk."""
+        for conf_key in FILE_MAPPING.values():
+            if not os.path.exists(self.hass.config.path(self.data[conf_key])):
+                return False
+        return True
+
+    @callback
+    def _configure_tls_assets(self):
+        """Fill the tls asset locations in self.data."""
+        for asset_key, conf_key in FILE_MAPPING.items():
+            self.data[conf_key] = TLS_ASSET_TEMPLATE.format(self.bridge_id, asset_key)
+
+    @callback
+    def _async_data_host_is_already_configured(self):
+        """Check to see if the host is already configured."""
+        return any(
+            self.data[CONF_HOST] == entry.data[CONF_HOST]
+            for entry in self._async_current_entries()
+            if CONF_HOST in entry.data
+        )
 
     async def async_step_import(self, import_info):
         """Import a new Caseta bridge as a config entry.
 
         This flow is triggered by `async_setup`.
         """
 
-        # Abort if existing entry with matching host exists.
         host = import_info[CONF_HOST]
-        if any(
-            host == entry.data[CONF_HOST] for entry in self._async_current_entries()
-        ):
-            return self.async_abort(reason=ABORT_REASON_ALREADY_CONFIGURED)
-
         # Store the imported config for other steps in this flow to access.
         self.data[CONF_HOST] = host
+
+        # Abort if existing entry with matching host exists.
+        if self._async_data_host_is_already_configured():
+            return self.async_abort(reason=ABORT_REASON_ALREADY_CONFIGURED)
```

Comment:
This is unconventional compared to most other config flows using strings directly

Reply:
Leaving this one as-is since its a style comment.